src/main/java/io/kokuwa/maven/helm/util/DependencyOverwriter.java

```diff
@@ -0,0 +1,233 @@
+		ObjectMapper yamlMapper = new YAMLMapper();
+		yamlMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+		Dependencies dependencies;
+		try {
+			dependencies = yamlMapper.readValue(chartFile, Dependencies.class);
+		} catch (IOException e) {
+			throw new MojoExecutionException("Unable to read chart dependencies from " + chartFile, e);
+		}
+
+		return dependencies.getDependencies();
+	}
+
+	/**
+	 * Sets the boolean for the {@link Dependency} that need to be overwritten
+	 * @param dependencies {@Link ArrayList}<{@link Dependency}>
+	 */
+	private void updateDependencies(ArrayList<Dependency> dependencies) {
+		for (Dependency dependency : dependencies) {
+			if (dependency.getRepository() != null && dependency.getRepository().startsWith("file://")) {
+				dependency.setOverwrite(true);
+			}
+		}
+	}
+
+	/**
+	 * Overwrite the dependencies with their new values to the Chart.yaml 
+	 * @param dependencies {@link ArrayList}<{@link Dependency}>
+	 * @param chartFile Chart.yaml file to read
+	 * @param directory Directory to write the new Chart.yaml to
+	 * @param overwriteRepository used to overwrite a local path chart's repository
+	 * @param overwriteVersion used to overwrite a local path chart's version
+	 * @param log
+	 * @throws MojoExecutionException
+	 */
+	private void overwriteDependencies(ArrayList<Dependency> dependencies, File chartFile, Path directory, 
+		String overwriteRepository, String overwriteVersion, Log log) throws MojoExecutionException {
+
+		//delete the outdated Chart.lock
+		File chartLock = new File(directory.toString() + "/Chart.lock");
+		
+		if (chartLock.exists()) {
+			chartLock.delete();
+			log.info("Chart.lock deleted at " + directory);
+		} else {
+			log.info("No Chart.lock file found at " + directory);
+		}
+
+		List<String> overwrittenChart = new ArrayList<>();
+		List<String> originalChart;
+		try {
+			originalChart = Files.readAllLines(Paths.get(chartFile.toURI()), StandardCharsets.UTF_8);
+		} catch (IOException e) {
+			throw new MojoExecutionException("Unable to read Chart.yaml at " + directory, e);
+		}
+
+		//Used for determining when a dependency has been read
+		boolean name, repository, version;
+		name = version = repository = false;
+
+		Boolean isDependencies = null;
+		Iterator<Dependency> dependencyIterator = dependencies.iterator();
+		Dependency currentDependency = dependencyIterator.next();
+
+		for (String line : originalChart) {
+
+			//all dependencies have been read so just need to write remaining lines
+			if (!dependencyIterator.hasNext() && isDependencies != null && !isDependencies) {
+				overwrittenChart.add(line);
+				continue;
+			}
+
+			//determines whether currently in the dependencies section of the yaml
+			if (line.contains("dependencies:")) {
+				isDependencies = true;
+				overwrittenChart.add(line);
+				continue;
+			}
+
+			//check for lines that need to be overwritten
+			if (isDependencies != null && isDependencies) {
```

Comment:
S: Treating `null` as a special value is usually an anti-pattern (unless it's at some integration point.)  You should be able to get around using `null` (and therefore just a simple boolean) by doing something like:
```
if( isDependencies )
    process dependency line
    if( current dependency finished && !dependencies.hasNext() )
        isDependencies = false
else if( line contains "dependencies:" )
    isDependencies = true

add line to overwritten chart
```

This also avoids needing to use the `continue` keyword which is nice from a readability/clarity standpoint